internal/services/authorization/role_assignment_resource.go

```diff
@@ -381,6 +382,17 @@ func retryRoleAssignmentsClient(d *pluginsdk.ResourceData, id parse.ScopedRoleAs
 				return pluginsdk.RetryableError(err)
 			case retryLinkedAuthorizationFailedError && response.WasForbidden(resp.HttpResponse) && strings.Contains(err.Error(), "LinkedAuthorizationFailed"):
 				return pluginsdk.RetryableError(err)
+			case response.WasStatusCode(resp.HttpResponse, http.StatusConflict) && strings.Contains(err.Error(), "RoleAssignmentExists"):
```

Comment:
Instead of checking here, shall we move it to L228 to do the existance check?

Reply:
After a reconsidering, if we are building two conflict role assignments, keeping the error handling here seems more reasonable, as otherwise there is a race condition and the latter assignment will still hit this conflict error, if we move the check to L228.